workflow/util/util.go

```diff
@@ -735,36 +735,60 @@ func convertNodeID(newWf *wfv1.Workflow, regex *regexp.Regexp, oldNodeID string,
 }
 
 // RetryWorkflow updates a workflow, deleting all failed steps as well as the onExit node (and children)
-func RetryWorkflow(ctx context.Context, kubeClient kubernetes.Interface, hydrator hydrator.Interface, wfClient v1alpha1.WorkflowInterface, name string, restartSuccessful bool, nodeFieldSelector string) (*wfv1.Workflow, error) {
+func RetryWorkflow(ctx context.Context, podIf v1.PodInterface, wfClient v1alpha1.WorkflowInterface, name string, restartSuccessful bool, nodeFieldSelector string) (*wfv1.Workflow, error) {
+	var updatedWf *wfv1.Workflow
+	err := waitutil.Backoff(retry.DefaultRetry, func() (bool, error) {
```

Comment:
I don't think we should do backoff deep in the stack. In fact, not at all in this case. The client will always need to implement some kind of retry themselves, so 2n retries will happen.